File: fixtures/scale/swift/medium/Functions/RoundingFunctions.swift
import Foundation

enum RoundingFunctions {
    static func install(_ registry: FunctionRegistry) {
        registry.define("round") { $0[0].rounded() }
        registry.define("floor") { $0[0].rounded(.down) }
        registry.define("ceil") { $0[0].rounded(.up) }
        registry.define("trunc") { $0[0].rounded(.towardZero) }
    }
}

File: fixtures/scale/java/medium/src/calc/funcs/RoundingFunctions.java
package calc.funcs;

import calc.eval.FunctionRegistry;

public class RoundingFunctions {
    public static void register(FunctionRegistry registry) {
        registry.addUnary("floor", Math::floor);
        registry.addUnary("ceil", Math::ceil);
        registry.addUnary("round", x -> (double) Math.round(x));
        registry.addUnary("trunc", x -> (double) (long) x);
        registry.addUnary("abs", Math::abs);
        registry.addUnary("sign", Math::signum);
    }
}
